lib/modules/neverEndingReddit.js

```diff
@@ -70,9 +70,31 @@ modules['neverEndingReddit'] = {
 	},
 	beforeLoad: function() {
 		if ((this.isEnabled()) && (this.isMatchURL())) {
-			RESTemplates.load('neverEndingReddit-CSS', function(template) {
-				RESUtils.addCSS(template.text());
+			return RESTemplates.load('neverEndingReddit-CSS', function(template) {
+				modules['neverEndingReddit']._css = template;
 			});
+		}
+	},
+	go: function() {
+		if ((this.isEnabled()) && (this.isMatchURL())) {
+			// code inspired by River of Reddit, but rewritten from scratch to work across multiple browsers...
+			// Original River of Reddit author: reddy kapil
+			// Original link to Chrome extension: https://chrome.google.com/extensions/detail/bjiggjllfebckflfdjbimogjieeghcpp
+
+			// store access to the siteTable div since that's where we'll append new data...
+			var stMultiCheck = document.querySelectorAll('#siteTable');
+			this.siteTable = stMultiCheck[0];
+			// stupid sponsored links create a second div with ID of sitetable (bad reddit! you should never have 2 IDs with the same name! naughty, naughty reddit!)
+			if (stMultiCheck.length === 2) {
+				// console.log('skipped first sitetable, stupid reddit.');
+				this.siteTable = stMultiCheck[1];
+			}
+			if (!this.siteTable) {
```

Comment:
This can be removed as soon as RES can handle the new post listing layout on the search page.